proteus-spring-core/src/main/java/io/netifi/proteus/spring/core/annotation/ProteusClientStaticFactory.java

```diff
@@ -46,32 +51,31 @@
     /**
      * Creates an instance of the correct Proteus client for injection into an annotated field.
      *
-     * @param clientClass proteus client class
+     * @param targetClass proteus client class
      * @return an instance of a Proteus client
      */
     public static Object getBeanInstance(
         final DefaultListableBeanFactory beanFactory,
-        final Class<?> clientClass,
+        final Class<?> targetClass,
         final ProteusClient proteusClientAnnotation
     ) {
-        final String beanName = getBeanName(proteusClientAnnotation, clientClass);
+        final String beanName = getBeanName(proteusClientAnnotation, targetClass);
 
         if (!beanFactory.containsBean(beanName)) {
             Proteus proteus = beanFactory.getBean(Proteus.class);
 
-            // Creating ProteusSocket Instance
-            ProteusSocket proteusSocket = null;
-
-            switch (proteusClientAnnotation.type()) {
-                case BROADCAST:
-                    proteusSocket = proteus.broadcast(proteusClientAnnotation.group());
-                    break;
-                case GROUP:
-                    proteusSocket = proteus.group(proteusClientAnnotation.group());
-                    break;
-                case DESTINATION:
-                    proteusSocket = proteus.destination(proteusClientAnnotation.destination(), proteusClientAnnotation.group());
-                    break;
+            //Tags reconciliation
+            TagSupplier tagSupplier = new NoTagsSupplier();
+            if(proteusClientAnnotation.tagSupplier() != null && !proteusClientAnnotation.tagSupplier().equals(NoTagsSupplier.class)){
```

Comment:
If you used a static instance could you check that it was the static instance?

Reply:
Here I can't because it needs a Class<TagSupplier> rather than an instance - the annotation defaults will only accept constants. I had to do this for a couple of the new fields, capture the Class rather than an instance.